### ISSUE TITLE
`createMemo` value not updating when signal dependency updates in Vitest

### ISSUE THREAD

**Issue by zzzachzzz** [solidjs/solid#1422]
### Describe the bug

When using `createMemo` in a unit test with Vitest, calling the returned memo signal does not yield the expected value after its dependency has updated.

```ts
it('memo updates when its signal dependency updates', () => {
  createRoot(() => {
    const [num, setNum] = createSignal(1);
    const plusOne = createMemo(() => num() + 1);
    const plusOneMore = createMemo(() => plusOne() + 1);

    expect(num()).toBe(1);
    expect(plusOne()).toBe(2);
    expect(plusOneMore()).toBe(3);

    setNum(2);

    expect(num()).toBe(2);
    expect(plusOne()).toBe(3); // AssertionError: expected 2 to be 3
    expect(plusOneMore()).toBe(4);
  });
});
```

What am I missing here? I'm not sure if there's something wrong with my test configuration. The example, with the test parts stripped out, works as expected in [this playground](https://playground.solidjs.com/anonymous/cda762ee-4c2e-40b8-abbf-0e3879093428).

### Your Example Website or App

N/A

### Steps to Reproduce the Bug or Issue

Run the above test code in Vitest (or Jest)?

### Expected behavior

I expect the memo signal to yield an updated value when its dependency updates.

### Platform

N/A

**Comment by lxsmnsyc** [comment]
if this is solid-js resolving to the server builds, this is expected. Server builds of solid-js doesn't have a reactive runtime. You might want to pull the browser builds through the `browser` import conditions.

**Comment by zzzachzzz** [comment]
Yep! Changing my import from `import ... from 'solid-js'` to `import ... from 'solid-js/dist/solid'` did the trick.

Do you know what might be causing it to resolve to the server builds? My Vite config is supposedly setup with `environment: 'jsdom'`.

Using the package [browser-or-node](https://www.npmjs.com/package/browser-or-node), I get the following output in a test:
```
isJsDom: true
isBrowser: true
isNode: true
isWebWorker: false
isDeno: false
```

**Comment by zzzachzzz** [comment]
```ts
// vite.config.ts
import { defineConfig } from 'vite';
import solidPlugin from 'vite-plugin-solid';
import tsconfigPaths from 'vite-tsconfig-paths';
import devtools from 'solid-devtools/vite';

export default defineConfig({
  plugins: [tsconfigPaths(), devtools(), solidPlugin()],
  server: {
    port: 3000,
  },
  build: {
    target: 'esnext',
  },
  test: {
    globals: true,
    environment: 'jsdom',
  },
});
```

**Comment by zzzachzzz** [comment]
Confirmed switching to jest and using a config with the `solid-jest` preset solves the issue as well:
```
module.exports = {
  preset: 'solid-jest/preset/browser',
};
```

**Comment by atk** [comment]
Can you try adding
```js
  test: {
    transformMode: { web: [/\.[jt]sx$/] }
  },
```
to your vite config?

**Comment by zzzachzzz** [comment]
@atk Same result with that vite config modification

**Comment by atk** [comment]
I forgot, for the normal plugin, you'll also need to define resolve conditions ["browser", "development"] so it won't think it's testing on the server even though using a jsdom environment.

**Comment by zzzachzzz** [comment]
Confirmed adding the following to the vite config solves the issue:
```ts
resolve: {
  conditions: ["browser", "development"],
}
```

`transformMode: { web: [/\.[jt]sx$/] }` did not appear to have an impact in my case.

**Comment by atk** [comment]
Thanks for the confirmation.
